### PR TITLE
[16.0][MIG] account_invoice_pricelist

### PR DESCRIPTION
**Pull request by hildickethan**
- Migration to 16.0
- Removed `l10n` folder since to my knowledge its redundant with the i18n folder, last update being 2015
- Cleaned up a lot of pricelist calculation code because it seems like base Odoo pricelist methods do most of the heavy lifting, but I believe all existing functionality remains
- Changed the placement of the Pricelist group in the search view, I don't think this is working in 13.0, 14.0 and 15.0 since I couldn't see the option on runboats until I went back to a 12.0

+ changes from @RodrigoBM https://github.com/OCA/account-invoicing/pull/1337#pullrequestreview-1448445513
Better adaptation of the code to V16.
Bug fixes
Remove the price list in the supplier invoice.
Added access to the price list from the finance menu in the customer section.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please include #1354

**Comment by hildickethan** [comment]
> Please include #1354

Included

File: account_invoice_pricelist/models/account_move.py
Comment (nguyenminhchien) [inline account_invoice_pricelist/models/account_move.py]: No need to call "self._set_pricelist_currency()" here. When you set value for self.pricelist_id at line 32, "_set_pricelist_currency" will be called as well.

File: account_invoice_pricelist/views/account_invoice_view.xml
Comment (nguyenminhchien) [inline account_invoice_pricelist/views/account_invoice_view.xml:7]: Should add the group "product.group_product_pricelist" for this button.
Same for the field, filter below

Comment (hildickethan) [reply account_invoice_pricelist/views/account_invoice_view.xml:7]: Good idea

**Comment by hildickethan** [comment]
@nguyenminhchien Comment changes pushed

File: account_invoice_pricelist/views/account_invoice_view.xml
Comment (nguyenminhchien) [inline account_invoice_pricelist/views/account_invoice_view.xml]: @hildickethan-S73 you may face an error when adding **groups="product.group_product_pricelist"** to this field.
Fix it by adding one more field:
`<field name="pricelist_id" groups="!product.group_product_pricelist" invisible="1"/>`

I learn this from [sale_order_views.xml#L279](https://github.com/odoo/odoo/blob/16.0/addons/sale/views/sale_order_views.xml#L279)

Comment (hildickethan) [reply account_invoice_pricelist/views/account_invoice_view.xml]: I skipped over this one by accident :sweat_smile: Yes it did error without the extra field because it is used in the domain above it. I added both so that the group covers all new fields now

**Review by nguyenminhchien — APPROVED** [review]
lgtm

**Comment by SoniaViciana** [comment]
LGTM!

**Review by RodrigoBM — APPROVED** [review]

**Review by SoniaViciana — APPROVED** [review]

**Review by RodrigoBM — CHANGES_REQUESTED** [review]
Hi @hildickethan-S73 , 

I have made you a  [PR](https://github.com/Studio73/account-invoicing/pull/1) in the branch. Improvements made:

- Better adaptation of the code to V16.
- Bug fixes
- Remove the price list in the supplier invoice.
- Added access to the price list from the finance menu in the customer section.

Regards

FYI: @nguyenminhchien

**Comment by RodrigoBM** [comment]
ping @hildickethan-S73

**Comment by hildickethan** [comment]
> Hi @hildickethan-S73 ,
> 
> I have made you a [PR](https://github.com/Studio73/account-invoicing/pull/1) in the branch. Improvements made:

Thanks for the changes, they LGTM

**Comment by RodrigoBM** [comment]
I will review the failed tests.

**Comment by RodrigoBM** [comment]
Hi @hildickethan-S73 

I have added this [PR](https://github.com/Studio73/account-invoicing/pull/3) which adds the following point.

    - test enabled configuration is added for the _check_currency method

As other tests in other modules may conflict with this restriction.

Sorry for the inconvenience

**Comment by RodrigoBM** [comment]
@hildickethan-S73 when you accept the PR you can squash or fixup my commit REF, thanks

**Comment by hildickethan** [comment]
> @hildickethan-S73 when you accept the PR you can squash or fixup my commit REF, thanks

I will when tests go green

**Review by RodrigoBM — APPROVED** [review]

**Comment by hildickethan** [comment]
rebased to 16.0 and squashed @RodrigoBM's commits together

**Comment by RodrigoBM** [comment]
is it possible to merge this PR? @OCA/accounting-maintainers

**Comment by RodrigoBM** [comment]
is it possible to merge this PR? @OCA/accounting-maintainers

**Review by AaronHForgeFlow — APPROVED** [review]
Functional

**Comment by legalsylvain** [comment]
Hi @hildickethan-S73. Thanks for porting this usefull module.

I have a problem with the test. Step to reproduce : 
- install the module in the fresh database with demo.
- stop the server
- run test : Test are failing.

Checking the error, I wonder if it's a problem related to https://github.com/OCA/account-invoicing/pull/1354 mentionned by @pedrobaeza.
Note : it is blocking other PR right now, like https://github.com/OCA/account-invoicing/pull/1436

Could you take a look ? 

thanks ! 

<details>
  <summary>See full logs</summary>
```
2023-06-11 22:29:16,981 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: ====================================================================== 
2023-06-11 22:29:16,981 24577 ERROR account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: FAIL: TestAccountMovePricelist.test_account_invoice_fixed_pricelist_without_discount_secondary_currency
Traceback (most recent call last):
  File "/home/sylvain/grap_dev/grap-odoo-env-16.0/src/OCA/account-invoicing__discount/account_invoice_pricelist/tests/test_account_move_pricelist.py", line 304, in test_account_invoice_fixed_pricelist_without_discount_secondary_currency
    self.assertAlmostEqual(invoice_line.price_unit, 65.41)
AssertionError: 100.0 != 65.41 within 7 places (34.59 difference)
 
2023-06-11 22:29:16,982 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: Starting TestAccountMovePricelist.test_account_invoice_pricelist ... 
2023-06-11 22:29:16,985 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: Starting TestAccountMovePricelist.test_account_invoice_pricelist_with_discount_secondary_currency ... 
2023-06-11 22:29:17,188 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: ====================================================================== 
2023-06-11 22:29:17,188 24577 ERROR account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: FAIL: TestAccountMovePricelist.test_account_invoice_pricelist_with_discount_secondary_currency
Traceback (most recent call last):
  File "/home/sylvain/grap_dev/grap-odoo-env-16.0/src/OCA/account-invoicing__discount/account_invoice_pricelist/tests/test_account_move_pricelist.py", line 283, in test_account_invoice_pricelist_with_discount_secondary_currency
    self.assertAlmostEqual(invoice_line.price_unit, 58.87)
AssertionError: 90.0 != 58.87 within 7 places (31.130000000000003 difference)
 
2023-06-11 22:29:17,190 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: Starting TestAccountMovePricelist.test_account_invoice_pricelist_without_discount ... 
2023-06-11 22:29:17,387 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: Starting TestAccountMovePricelist.test_account_invoice_pricelist_without_discount_secondary_currency ... 
2023-06-11 22:29:17,636 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: ====================================================================== 
2023-06-11 22:29:17,636 24577 ERROR account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: FAIL: TestAccountMovePricelist.test_account_invoice_pricelist_without_discount_secondary_currency
Traceback (most recent call last):
  File "/home/sylvain/grap_dev/grap-odoo-env-16.0/src/OCA/account-invoicing__discount/account_invoice_pricelist/tests/test_account_move_pricelist.py", line 290, in test_account_invoice_pricelist_without_discount_secondary_currency
    self.assertAlmostEqual(invoice_line.price_unit, 65.41)
AssertionError: 100.0 != 65.41 within 7 places (34.59 difference)
 
2023-06-11 22:29:17,638 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: Starting TestAccountMovePricelist.test_account_invoice_with_discount_change_pricelist ... 
2023-06-11 22:29:17,774 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: Starting TestAccountMovePricelist.test_account_invoice_without_discount_change_pricelist ... 
2023-06-11 22:29:17,917 24577 INFO account_invoice_pricelist_16_test odoo.addons.account_invoice_pricelist.tests.test_account_move_pricelist: Starting TestAccountMovePricelist.test_check_currency ... 
2023-06-11 22:29:18,090 24577 INFO account_invoice_pricelist_16_test odoo.addons.base.models.ir_attachment: filestore gc 0 checked, 0 removed 
2023-06-11 22:29:18,122 24577 INFO account_invoice_pricelist_16_test odoo.modules.loading: Module account_invoice_pricelist loaded in 2.64s (incl. 2.19s test), 140 queries (+898 test, +140 other) 
2023-06-11 22:29:18,123 24577 ERROR account_invoice_pricelist_16_test odoo.modules.loading: Module account_invoice_pricelist: 3 failures, 0 errors of 10 tests 

```
</details>

**Comment by hildickethan** [comment]
@legalsylvain I will try to look at it within the next few days, but at a quick glance I don't any other PR being blocked by this, it could be a compatibility issue between the 2 modules that needs updating to 16.0

**Pull request merged**
